src/features/script-load.js

```diff
@@ -14,9 +14,47 @@ systemJSPrototype.createScript = function (url) {
   return script;
 };
 
+// Auto imports -> script tags can be inlined directly for load phase
+var lastAutoImportUrl, lastAutoImportDeps;
+var autoImportCandidates = {};
+var systemRegister = systemJSPrototype.register;
+var timeoutCnt = 0;
+systemJSPrototype.register = function (deps, declare) {
+  if (hasDocument && document.readyState === 'loading' && typeof deps !== 'string') {
+    var scripts = document.getElementsByTagName('script');
+    var lastScript = scripts[scripts.length - 1];
+    var url = lastScript && lastScript.src;
+    if (url) {
+      lastAutoImportUrl = url;
+      lastAutoImportDeps = deps;
+      autoImportCandidates[url] = [deps, declare];
+      var loader = this;
+      // This timeout ensures that if this is a dynamic script injection by SystemJS
+      // that the auto import will be cleared after the timeout and hence will not
+      // be auto imported
+      timeoutCnt++;
+      setTimeout(function () {
+        if (autoImportCandidates[url])
+          loader.import(url);
+        if (--timeoutCnt === 0 && document.readyState !== 'loading')
+          autoImportCandidates = {};
+      }, 0);
```

Comment:
```suggestion
      });
```

The saved bytes :)

Reply:
Good catch. Does this work reliably in older browsers?